File: test/java/junit5-to-testng/parameterized/PARAM-001.expected.java
import org.testng.Assert;

public class ValueSourceTest {
    // TERRAIN-TODO [UNCONVERTIBLE-PARAMETERIZED-TEST]: JUnit 5 @ParameterizedTest requires manual conversion to TestNG @DataProvider
    // Original: @ParameterizedTest
    // Manual action required: Create a @DataProvider method and reference it with @Test(dataProvider = "...")
    @ParameterizedTest
    // TERRAIN-TODO [UNCONVERTIBLE-VALUE-SOURCE]: JUnit 5 @ValueSource has no direct TestNG equivalent
    // Original: @ValueSource(strings = {"hello", "world"})
    // Manual action required: Convert values into a @DataProvider method returning Object[][]
    @ValueSource(strings = {"hello", "world"})
    public void testStrings(String value) {
        Assert.assertNotNull(value);
    }
}

File: test/python/pytest-to-unittest/parametrize/PARAM-001.expected.py
import pytest
import unittest


class TestIncrement(unittest.TestCase):
    # HAMLET-TODO [UNCONVERTIBLE-PARAMETRIZE]: @pytest.mark.parametrize has no direct unittest equivalent
    # Original: @pytest.mark.parametrize("x,expected", [(1, 2), (2, 3)])
    # Manual action required: Use subTest() or create individual test methods for each parameter set
    @pytest.mark.parametrize("x,expected", [(1, 2), (2, 3)])
    def test_increment(self, x, expected):
        self.assertEqual(x + 1, expected)
